tests/test_install.py

```diff
@@ -0,0 +1,308 @@
+    credentials_dict = three_nodes_config_dict.get('credentials')
+    _iterate_nested_dict(credentials_dict, chosen_keys, fixed_value)
+    _populate_credentials(credentials_dict)
+    _assert_dict_values_not_none(credentials_dict)
+    _iterate_nested_dict(credentials_dict, chosen_keys, fixed_value, True)
+
+
+def test_config_files_credentials(three_nodes_config_dict, config_files_dir):
+    """Test the credentials in the config files are populated correctly."""
+    # In this case, The three nodes and nine nodes logic is the same
+    credentials_dict = three_nodes_config_dict.get('credentials')
+    _populate_credentials(credentials_dict)
+    _create_config_files(three_nodes_config_dict, config_files_dir,
+                         credentials_dict)
+    _assert_manager_config_credentials(config_files_dir, credentials_dict)
+    _assert_postgresql_config_credentials(config_files_dir, credentials_dict)
+    _assert_rabbitmq_config_credentials(config_files_dir, credentials_dict)
+
+
+def test_ldap_in_config_file(three_nodes_config_dict,
+                             config_files_dir,
+                             ldap_ca_path,
+                             tmp_certs_dir,
+                             cluster_manager_certs_dir):
+    """Test if LDAP is configured properly in the manager config.yaml file."""
+    # In this case, The three nodes and nine nodes logic is the same
+    cluster_manager_ldap_ca = str(cluster_manager_certs_dir / 'ldap_ca.pem')
+    ldap_dict = {
+        'server': 'ldaps://1.1.1.1',
+        'domain': 'test_domain',
+        'is_active_directory': True,
+        'ca_cert': ldap_ca_path,
+        'username': 'test_username',
+        'password': 'test_password',
+        'dn_extra': 'test_dn_extra'
+    }
+    three_nodes_config_dict.update({'ldap': copy.deepcopy(ldap_dict)})
+
+    with mock.patch('cfy_cluster_manager.main.LDAP_CA_PATH',
+                    cluster_manager_ldap_ca):
+        _handle_certificates(three_nodes_config_dict, None)
+        _create_config_files(three_nodes_config_dict, config_files_dir)
+    manager_config = _get_instance_config('manager', config_files_dir)
+
+    ldap_dict['ca_cert'] = cluster_manager_ldap_ca
+    assert manager_config['restservice']['ldap'] == ldap_dict
+    _assert_created_certs(tmp_certs_dir, cluster_manager_certs_dir)
+
+
+def test_extrnal_db_in_config_file(three_nodes_external_db_config_dict,
+                                   config_files_dir,
+                                   external_db_ca_path,
+                                   tmp_certs_dir,
+                                   cluster_manager_certs_dir):
+    """
+    Test if the external_db is configured properly in the manager
+    config.yaml file.
+    """
+    # In this case, The three nodes and nine nodes logic is the same
+    cluster_manager_external_db_ca = str(
+        cluster_manager_certs_dir / 'external_db_ca.pem')
+    external_db_config = {
+        'host': 'user.postgres.database.azure.com',
```

Comment:
.example or .test rather than .com would be better- best to avoid using (potentially) real subdomains.

Reply:
Same as the comment above. Great catch.